### PR TITLE
KEP-4193: Promote ServiceAccountTokenJTI, ServiceAccountTokenPodNodeInfo, ServiceAccountTokenNodeBindingValidation to stable

### PR DESCRIPTION
**Pull request by liggitt**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Promotes three feature gates for https://github.com/kubernetes/enhancements/issues/4193 to stable

Adds credential-id / pod / node claim verification to the e2e conformance test.

#### Does this PR introduce a user-facing change?
```release-note
* Promotes the ServiceAccountTokenJTI feature to GA, which adds a `jti` claim to issued service account tokens and embeds the `jti` claim as a `authentication.kubernetes.io/credential-id=["JTI=..."]` value in user extra info
* Promotes the ServiceAccountTokenPodNodeInfo feature to GA, which adds the node name and uid as claims into service account tokens mounted into running pods, and embeds that information as `authentication.kubernetes.io/node-name` and `authentication.kubernetes.io/node-uid` user extra info when the token is used
* Promotes the ServiceAccountTokenNodeBindingValidation feature to GA, which validates service account tokens bound directly to nodes.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/issues/4193
```

/sig auth
/cc @munnerz @deads2k

### CODE REVIEW

**Comment by liggitt** [comment]
> ERROR: failed to solve: registry.k8s.io/build-image/kube-cross:v1.32.0-go1.23.2-bullseye.0: failed to resolve source metadata for registry.k8s.io/build-image/kube-cross:v1.32.0-go1.23.2-bullseye.0: unexpected status from HEAD request to https://us-central1-docker.pkg.dev/v2/k8s-artifacts-prod/images/build-image/kube-cross/manifests/v1.32.0-go1.23.2-bullseye.0: 429 Too Many Requests

**Review by munnerz — APPROVED** [review]
Thanks for putting this together & the additional end-to-end tests too!

I am guessing we cannot promote ServiceAccountTokenNodeBinding to GA until v1.33, as some users could be running their cluster with the `ServiceAccountTokenNodeBindingValidation` feature gate *disabled*.

Looking at https://github.com/kubernetes/enhancements/tree/e1d4725d07a07c40da7ded6e1dd44c0f9aaeaeb9/keps/sig-auth/4193-bound-service-account-token-improvements#rollout-upgrade-and-rollback-planning, I think we may need to update some of the details on rollout/rollback to align with these version numbers? (as we are a version out it looks like?).

I'll put together a PR to adjust the KEP if so 👀 

/lgtm
/hold in case you want anyone else in particular to review before this merges :)

**Pull request merged**
